checkov/circleci_pipelines/runner.py

```diff
@@ -45,6 +48,34 @@ def is_workflow_file(self, file_path: str) -> bool:
         abspath = os.path.abspath(file_path)
         return WORKFLOW_DIRECTORY in abspath and abspath.endswith(("config.yml", "config.yaml"))
 
+    def get_resource(self, file_path: str, key: str, supported_entities: Iterable[str],
+                     start_line: int = -1, end_line: int = -1) -> str:
```

Comment:
This function implicitly assumes that `len(supported_entities) = 1`
Currently, every circleci pipelines check has a single supported entity, but in the future, this might not be true.

I think we should check that and add a warning log so that the assumption will be explicit and it will make it easier to debug in the future

Reply:
NICE! 🍒 but actually if there are no supported entities, the returned key is actually the input key. I think this is valid for all implemented cases (line 59, 77 incircleCI)

Reply:
I was worried about the case where a check can have multiple `supported_entities`
Then the order of the if clauses will determine it's resource id

Reply:
🤯 - added a comment, LMK if this is enough